src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -191,85 +242,85 @@ object ApiOperations {
   // Predicates to determine existence of specific type of filter
 
   private def isBlockFilter(filter: Filter): Boolean =
-    (filter.blockIDs.isDefined && filter.blockIDs.get.nonEmpty) ||
-      (filter.levels.isDefined && filter.levels.get.nonEmpty) ||
-      (filter.chainIDs.isDefined && filter.chainIDs.get.nonEmpty) ||
-      (filter.protocols.isDefined && filter.protocols.get.nonEmpty)
+    filter.blockIDs.nonEmpty ||
+      filter.levels.nonEmpty ||
+      filter.chainIDs.nonEmpty ||
+      filter.protocols.nonEmpty
 
   private def isOperationGroupFilter(filter: Filter): Boolean =
-    (filter.operationGroupIDs.isDefined && filter.operationGroupIDs.get.nonEmpty) ||
-      (filter.operationSources.isDefined && filter.operationSources.get.nonEmpty)
+    filter.operationGroupIDs.nonEmpty ||
```

Comment:
would it be perhaps helpful to make a single "isXFilter" function, because besides slight variations, the majority of the logic here is repeated? Or do you think that would be over engineering it?

Reply:
indeed, but it's already done in one of the other pending PRs, so I'm reluctant to refactor here again, I'd solve the conflict when the PRs are merged